src/services/hooks/queries/useUsers.ts

```diff
@@ -0,0 +1,21 @@
+import { useQuery } from "@tanstack/react-query";
+import { getAllUsers, getSingleUser } from "../../apis/users";
+import { FetchUsersQuery } from "@/types/users";
+import { GET_USER, GET_USERS } from "@/constants/queryKeys";
+
+export const useGetAllUsers = <T>(query: FetchUsersQuery) => {
+  return useQuery({
+    queryKey: [GET_USERS, query],
+    queryFn: () => getAllUsers(query),
+    select: (res) => res.data as T,
+  });
+};
+
+export const useGetSingleUser = <T>({ user_id }: { user_id: string }) => {
```

Comment:
@AdejokeOgunyinka You don't have to pass an object if it only has one property, just pass an argument as a string instead.